### PR TITLE
Add ngl component 

### PR DESCRIPTION
**Pull request by SarahAlidoost**
In this pull request, a component is created to replace the ngl viewer parts of the haddock3 analyse module.

### CODE REVIEW

File: src/components/NglViewer/NglViewer.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { useEffect, useRef } from "react";
+import * as NGL from "ngl";
```
Comment (sverhoeven) [inline src/components/NglViewer/NglViewer.tsx:2]: For easier tree shaking do not use `* as ...` imports.

```suggestion
import {Stage, type Component} from "ngl";
```

File: src/components/NglViewer/NglViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  useEffect(() => {
+    if (stage.current === null) {
+      return;
+    }
+    if (activeStructure.fileName === "") {
+      dialogRef.current?.close();
+    } else {
+      dialogRef.current?.showModal();
+      stage.current.removeAllComponents();
```
Comment (sverhoeven) [inline src/components/NglViewer/NglViewer.tsx:33]: This line fixes https://github.com/haddocking/haddock3/issues/670, when https://github.com/haddocking/haddock3/pull/683 on that repo is merged we can close it.

File: src/components/NglViewer/NglViewer.css [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.viewport {
+  width: 400px;
+  height: 400px;
+}
+
+.dl {
+  position: absolute;
```
Comment (sverhoeven) [inline src/components/NglViewer/NglViewer.css:7]: The download link and close button are styled so the structure is shown underneath.
 
localhost_5173_

Its nice that it maximises the viewport, but can make the file name hard to read.

File: src/components/NglViewer/NglViewer.css [hunk 4]
```diff
@@ -0,0 +1,20 @@
+.viewport {
+  width: 400px;
+  height: 400px;
```
Comment (sverhoeven) [inline src/components/NglViewer/NglViewer.css:3]: The structure viewer is a bit small. It does allow you to see the table behind it, but I would like it to be a bigger.
```suggestion
  width: 800px;
  height: 600px;
```

**Review by sverhoeven — APPROVED** [review]
Nice work.

I was able to see structures with `npm run dev` and in `example.local.html`.

I just have some small inline suggestions.

File: src/examples/target.pdb
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply src/examples/target.pdb:1]: added.

File: src/components/NglViewer/NglViewer.css (hunk 3 above)
Comment (SarahAlidoost) [reply src/components/NglViewer/NglViewer.css:7]: I dont see the same problem, it seems the link and the view are separate :thinking: 
image

Comment (SarahAlidoost) [reply src/components/NglViewer/NglViewer.css:7]: the problem occurs in the local example html file
image

File: src/components/NglViewer/NglViewer.tsx (hunk 2 above)
Comment (SarahAlidoost) [reply src/components/NglViewer/NglViewer.tsx:33]: tested with two different files, the issue should be fixed.

File: src/components/NglViewer/NglViewer.css (hunk 3 above)
Comment (SarahAlidoost) [reply src/components/NglViewer/NglViewer.css:10]: changing the positions in css file should fix it.

**Pull request merged**